packages/kit/src/runtime/client/router.js

```diff
@@ -157,16 +157,16 @@ export class Router {
 
 			if (!this.owns(url)) return;
 
-			const noscroll = a.hasAttribute('sveltekit:noscroll');
+			// Check if new url only differs by hash
+			if (url.href.split('#')[0] === location.href.split('#')[0]) {
```

Comment:
`(some url).href.split('#')[0]` -> `(some url).hash`? (It will include the leading `#`)